velox/functions/sparksql/Arithmetic.h

```diff
@@ -350,4 +350,59 @@ struct ToHexBigintFunction {
     ToHexUtil::toHex(input, result);
   }
 };
+
+namespace detail {
+FOLLY_ALWAYS_INLINE static int8_t fromHex(char c) {
+  if (c >= '0' && c <= '9') {
+    return c - '0';
+  }
+
+  if (c >= 'A' && c <= 'F') {
+    return 10 + c - 'A';
+  }
+
+  if (c >= 'a' && c <= 'f') {
+    return 10 + c - 'a';
+  }
+  return -1;
+}
+} // namespace detail
+
+template <typename T>
+struct UnHexFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Varbinary>& result,
+      const arg_type<Varchar>& input) {
+    const auto resultSize = (input.size() + 1) >> 1;
+    result.resize(resultSize);
+    const char* inputBuffer = input.data();
+    char* resultBuffer = result.data();
+
+    int32_t i = 0;
+    if ((input.size() & 0x01) != 0) {
```

Comment:
This is so confusing. 

If input size is 1, then we decode the only character and return decoded calue or null if not valid.

If input size is > 1 and odd, we skip first character. Do we still need to decode that character and return null if it is not valid?

When skipping first character, do we append '0' at the end of the result? Is this how Spark behaves? I don't think this is described in the docs.

Reply:
> If input size is > 1 and odd, we skip first character. 

we do not skip first character, 

> Do we still need to decode that character and return null if it is not valid?

yes.

Reply:
> When skipping first character, do we append '0' at the end of the result? Is this how Spark behaves?

when first byte is valid, we skip it. yes, it's Spark's behaviors.